File: interviewplannerapp_ui/src/services/InterviewerService.js
import http from "../http-common"; 

class InterviewerService {
  getAllInterviewers(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/interviewer/interviewers`, searchDTO);
  }

  get(interviewerId) {
    return this.getRequest(`/interviewer/${interviewerId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/interviewer?field=${matchData}`, null);
  }

  addInterviewer(data) {
    return http.post("/interviewer/addInterviewer", data);
  }

  update(data) {
  	return http.post("/interviewer/updateInterviewer", data);
  }
  
  uploadImage(data,interviewerId) {
  	return http.postForm("/interviewer/uploadImage/"+interviewerId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new InterviewerService();

File: src/main/java/com/interviewplannerapp/service/InterviewerService.java
package com.interviewplannerapp.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.interviewplannerapp.domain.Interviewer;
import com.interviewplannerapp.dto.InterviewerDTO;
import com.interviewplannerapp.dto.InterviewerSearchDTO;
import com.interviewplannerapp.dto.InterviewerPageDTO;
import com.interviewplannerapp.dto.InterviewerConvertCriteriaDTO;
import com.interviewplannerapp.service.GenericService;
import com.interviewplannerapp.dto.common.RequestDTO;
import com.interviewplannerapp.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface InterviewerService extends GenericService<Interviewer, Integer> {

	List<Interviewer> findAll();

	ResultDTO addInterviewer(InterviewerDTO interviewerDTO, RequestDTO requestDTO);

	ResultDTO updateInterviewer(InterviewerDTO interviewerDTO, RequestDTO requestDTO);

    Page<Interviewer> getAllInterviewers(Pageable pageable);

    Page<Interviewer> getAllInterviewers(Specification<Interviewer> spec, Pageable pageable);

	ResponseEntity<InterviewerPageDTO> getInterviewers(InterviewerSearchDTO interviewerSearchDTO);
	
	List<InterviewerDTO> convertInterviewersToInterviewerDTOs(List<Interviewer> interviewers, InterviewerConvertCriteriaDTO convertCriteria);

	InterviewerDTO getInterviewerDTOById(Integer interviewerId);







}





